crates/compiler/load_internal/src/file.rs

```diff
@@ -2161,11 +2161,25 @@ fn report_unused_imported_modules(
         Occupied(entry) => entry.into_mut(),
     };
 
-    for (unused, region) in unused_imported_modules.drain() {
-        if !unused.is_builtin() {
-            existing.push(roc_problem::can::Problem::UnusedModuleImport(
-                unused, region,
-            ));
+    // TODO this outer conditional can be replaced by just the for loop
+    // once we have Task as builtin. (Also the for loop doesn't need the "Task" check.)
+    if !unused_imported_modules.is_empty() {
```

Comment:
Why is the outer conditional needed at all? Just to avoid locking module_ids?

Reply:
Yeah exactly - almost always there will be 0 unused imported modules, so I wanted to avoid unnecessary contention.